Tests/SwiftParserTest/translated/TypeExprTests.swift

```diff
@@ -49,11 +34,7 @@ final class TypeExprTests: XCTestCase {
         static func meth() {} 
         func instMeth() {} 
       }
-      """,
-      diagnostics: [
-        // TODO: Old parser expected error on line 5: protocol methods must not have bodies
-        // TODO: Old parser expected error on line 6: protocol methods must not have bodies
-      ]
+      """
```

Comment:
Why was this removed?

Reply:
SwiftSyntax uses the same nodes for functions inside protocols and structs. Since protocols have optional bodies, function bodies are also optional for functions in in structs in SwiftSyntax. This has two advantages:
- It reduces the nodes we need in SwiftSyntax (we only need a single one and not one for protocols and one for all the other types)
- `.swiftinterface` files also have optional function bodies (even for functions in structs). This way, we can parse `.swiftinterface` files without any problems.

The idea is that the missing function body will be diagnosed by ASTGen further down in the compilation pipeline instead of by the parser.

Reply:
I appreciate the detailed clarification!